x/concentrated-liquidity/tick.go

```diff
@@ -63,13 +63,31 @@ func (k Keeper) crossTick(ctx sdk.Context, poolId uint64, tickIndex int64, swapS
 		return sdk.Dec{}, err
 	}
 
-	accum, err := k.getFeeAccumulator(ctx, poolId)
+	feeAccum, err := k.getFeeAccumulator(ctx, poolId)
 	if err != nil {
 		return sdk.Dec{}, err
 	}
 
 	// subtract tick's fee growth outside from current fee growth global, including the fee growth of the current swap.
-	tickInfo.FeeGrowthOutside = accum.GetValue().Add(swapStateFeeGrowth).Sub(tickInfo.FeeGrowthOutside)
+	tickInfo.FeeGrowthOutside = feeAccum.GetValue().Add(swapStateFeeGrowth).Sub(tickInfo.FeeGrowthOutside)
+
+	uptimeAccums, err := k.getUptimeAccumulators(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	// Update global accums to now before uptime outside changes
+	if err := k.updateUptimeAccumulatorsToNow(ctx, poolId); err != nil {
```

Comment:
Why do we need to update them as we cross them?

Reply:
Shouldn't we be able to update them once per block?

Reply:
My thinking was that the update logic for the target tick's uptime accumulators will involve the global accum values, so we need to ensure they are up to date. Are you suggesting this isn't necessary?

I do see a different issue though – we should be updating the accums prior to getting them instead of after, will fix.

Reply:
So over a single swap, we might be crossing multiple ticks. From my understanding, we do not need to update the uptime accumulators every tick because that is idempotent within a swap. As a result, calling this update function as we cross every tick might be redundant.

Reply:
Outside of the scope of this PR but - why do we also need to call `updateUptimeAccumulatorsToNow` in `getTickInfo`?

It feels that this update needs to happen at another level of abstraction or in another location

Reply:
> So over a single swap, we might be crossing multiple ticks. From my understanding, we do not need to update the uptime accumulators every tick because that is idempotent within a swap. As a result, calling this update function as we cross every tick might be redundant.

Good point – we can add a check so it only updates if `LastUpdateTime < current time`. Made an issue here (#4422) to not block this PR (and because it could break some downstream PRs)

Reply:
> Outside of the scope of this PR but - why do we also need to call `updateUptimeAccumulatorsToNow` in `getTickInfo`?
> 
> It feels that this update needs to happen at another level of abstraction or in another location

We update there simply because `getInitialUptimeGrowthOutsidesForTick` uses global accum values (so they need to be up to date). The level of abstraction does seem a little off, but the function does include accumulator init logic more broadly so it might be worth discussing a small refactor or rename